initializr-generator/src/main/groovy/io/spring/initializr/metadata/DependenciesCapability.groovy

```diff
@@ -56,7 +56,18 @@ class DependenciesCapability extends ServiceCapability<List<DependencyGroup>> {
 	@Override
 	void merge(List<DependencyGroup> otherContent) {
 		otherContent.each { group ->
-			if (!content.find { group.name.equals(it.name)}) {
+			def contentGroup = content.find { group.name.equals(it.name)}
+			if (contentGroup != null) {
+				contentGroup.bom = group.bom
+				contentGroup.repository = group.repository
+				contentGroup.versionRange = group.versionRange
+				group.content.each { dependency ->
```

Comment:
That's a bit hard, isn't it? If you attempt to merge one group with another, all the bom/repository/versionRange are overridden (even if the new group has  null values there). The merge algorithm is pretty consistent until now: if you have an element with ID that doesn't exist, we'll add that. Granted, we're not merging dependencies within an existing group and that's an interesting use case (the code should move in a new `merge` method on `DependencyGroup` then). I am not sure I want to override bom/repo/versionRange.